app/src/main/java/com/teamagam/gimelgimel/app/sensor/viewModel/adapter/SensorRecyclerArrayAdapter.java

```diff
@@ -0,0 +1,112 @@
+package com.teamagam.gimelgimel.app.sensor.viewModel.adapter;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.teamagam.gimelgimel.R;
+import com.teamagam.gimelgimel.app.common.base.adapters.BaseRecyclerArrayAdapter;
+import com.teamagam.gimelgimel.app.common.base.adapters.BaseRecyclerViewHolder;
+import com.teamagam.gimelgimel.app.common.base.adapters.DataRandomAccessor;
+import com.teamagam.gimelgimel.app.sensor.model.SensorMetadataApp;
+
+import butterknife.BindView;
+
+public class SensorRecyclerArrayAdapter extends
+        BaseRecyclerArrayAdapter<SensorRecyclerArrayAdapter.SensorRecyclerViewHolder, SensorMetadataApp> {
+
+
+    private Drawable mSensorDrawable;
+    private SensorMetadataApp mLastSelected;
+    private DataRandomAccessor<SensorMetadataApp> mAccessor;
+
+    public SensorRecyclerArrayAdapter(
+            DataRandomAccessor<SensorMetadataApp> data,
+            OnItemClickListener<SensorMetadataApp> listener) {
+        super(data, listener);
+        mAccessor = data;
+    }
+
+    public synchronized void select(String sensorId) {
+        unselectLast();
+        selectNew(sensorId);
+        notifyDataSetChanged();
+    }
+
+    @Override
+    protected SensorRecyclerViewHolder createNewViewHolder(View v) {
+        return new SensorRecyclerViewHolder(v);
+    }
+
+    @Override
+    protected int getSingleItemLayoutRes() {
+        return R.layout.recycler_sensor_list_item;
+    }
+
+    @Override
+    protected void bindItemToView(SensorRecyclerViewHolder holder,
+                                  SensorMetadataApp sensorMetadataApp) {
+        setItemDrawable(holder);
+
+        holder.nameTextView.setText(sensorMetadataApp.getName());
+
+        setBackgroundColor(holder, sensorMetadataApp);
+    }
+
+    private void unselectLast() {
+        if (mLastSelected != null) {
+            mLastSelected.unselect();
+            mLastSelected = null;
```

Comment:
I'm doing it to avoid leaving the object in a malfunctioned state.
If anyone will use unselectLast somewhere else in the code after refactoring it or whatever, I'd like to know that mLastSelected really points to the last selected message